src/QsCompiler/DataStructures/SyntaxTokens.fs

```diff
@@ -56,6 +56,19 @@ type CharacteristicsKind<'S> =
 | Union of 'S * 'S
 | Intersection of 'S * 'S
 | InvalidSetExpr
+    with
+    member this.TryGetSimpleSet(simpleSet: OpProperty byref) =
```

Comment:
What is the reason for wanting to provide these TryGet* methods instead of using something like `value as SimpleSet set` (assuming they are primarily for use within C#)?

Reply:
I think there is no strong reason to have the TryGet* methods. I will change it to use the `is` pattern instead.

Reply:
Removed these methods since the pattern you suggested no longer needs them.